src/generate/register.rs

```diff
@@ -476,33 +558,74 @@ pub fn fields(
             } else {
                 quote! {
                     ///Writes raw bits to the field
-                    #[inline(always)]
+                    #inline
                     pub #unsafety fn #bits(self, value: #fty) -> &'a mut W {
                         self.w.bits = (self.w.bits & !#hexmask) | ((value as #rty) & #hexmask);
                         self.w
                     }
                 }
             });
 
-            let doc = format!("Write proxy for field `{}`", f.name);
+            let (doc, offset_entry) = if let Some((_, _, _, _, suffixes_str)) = &field_dim {
+                (
+                    format!("Write proxy for fields `{}{}`", &name, &suffixes_str),
+                    quote! {offset: usize,},
+                )
+            } else {
+                (format!("Write proxy for field `{}`", f.name), quote! {})
+            };
```

Comment:
You can use the following syntax to make things nicer:
```rust
let doc;
let offset_entry;
if ... {
    doc = format!(...);
    offset_entry = quote! {...};
} else {
    doc = format!(...);
    offset_entry = quote! {};
}
```

Reply:
Done.